omod-common/src/main/java/org/openmrs/module/webservices/rest/web/api/impl/RestServiceImpl.java

```diff
@@ -632,16 +632,23 @@ private void eliminateCandidateSearchHandlersWithMissingRequiredParameters(Set<S
 		return resourceHandlers;
 	}
 	
-	/* (non-Javadoc)
-	 * @see org.openmrs.module.webservices.rest.web.api.RestService#getSearchHandlers(java.lang.String)
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.api.RestService#getAllSearchHandlers()
+	 * @should return all search handlers if search handlers have been initialized
```

Comment:
Its true that some logic is really specific to an implementation and this is why we have a few tests like XXServiceImplTest or even there are situations where the logic really is DAO specific hence we end up with XXDAOTest classes. But in this case I would think that behavior is part of the service contract.